melodies_monet/util/sat_l2_swath_utility_tempo.py

```diff
@@ -0,0 +1,899 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+
+# read all swath data for the time range
+# developed for TEMPO Level2 NO2
+#
+
+"""Python utility for TEMPO use."""
+
+import collections
+import glob
+import logging
+import warnings
+
+import cf_xarray as cfxr
+import monetio as mio
+import numba
+import numpy as np
+import xarray as xr
+import xesmf as xe
+
+numba_logger = logging.getLogger("numba")
+numba_logger.setLevel(logging.WARNING)
+
+
+def calc_grid_corners(ds, lat="latitude", lon="longitude"):
+    """Adds latitude and longitude bounds inplace.
+    If the grid is rectilinear, it should be quite precise.
+    If it is curvilinear, is a rough estimate.
+
+    Parameters
+    ----------
+    ds : xr.Dataset
+        Dataset to which the latitude and longitude will be added.
+    lat : str
+        name of the lat variable.
+    lon : str
+        name of the lon variable.
+
+    Returns
+    -------
+    None
+    """
+    corners = ds[[lat, lon]].cf.add_bounds([lat, lon])
+    lat_b = cfxr.bounds_to_vertices(corners[f"{lat}_bounds"], "bounds", order=None)
+    lon_b = cfxr.bounds_to_vertices(corners[f"{lon}_bounds"], "bounds", order=None)
+    return lat_b, lon_b
```

Comment:
Function description indicates that these should be added in place. 
```suggestion
    ds['lat_b'] = cfxr.bounds_to_vertices(corners[f"{lat}_bounds"], "bounds", order=None)
    ds['lon_b'] = cfxr.bounds_to_vertices(corners[f"{lon}_bounds"], "bounds", order=None)
    return 
```

This is also a function that we'll want to use more broadly within the satellite tool in order to use xesmf's conservative regridding method.

Reply:
OK. Keeping inplace.